File: 2DAM/Desarrollo De Interfaces/tEMA3-deint/Ejercicio1-ExamenDEINT-T3/Ejercicio1-ExamenDEINT-T3/Clase/Tarea.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ejercicio1_ExamenDEINT_T3.Clase
{
    internal class Tarea
    {
        public string Nombre { get; set; }
        public string Descripcion { get; set; }
        public DateTime Fecha_vencimiento { get; set; }
        public string Estado { get; set; }

        public Tarea() { }

        public Tarea(string nombre, string descripcion, DateTime fecha_vencimiento, string estado)
        {
            Nombre = nombre;
            Descripcion = descripcion;
            Fecha_vencimiento = fecha_vencimiento;
            Estado = estado;
        }
    }
}

File: 2DAM/Sistema de Gestión Empresarial/Odoo/Ejercicio4/GestorTareas/models/tarea.py
from odoo import models, fields, api

class Tarea(models.Model):
    _name = "tareas.tarea"
    _description = "Gestor de Tareas - Tareas"

    nombre = fields.Char('Nombre', required=True)
    descripcion = fields.Text('Descripcion')
    fecha_creacion = fields.Datetime('Fecha Creacion', default=lambda self: fields.Datetime.now())
    contador_personas_asignadas = fields.Integer(compute='_compute_total')
    personas_ids = fields.Many2many('tarea.persona')
    estado = fields.Selection([
        ('porHacer', 'Por Hacer'),
        ('enProgreso', 'En Progreso'),
        ('hecho', 'Hecho')],
        string='Estado', default='porHacer', required=True)

    @api.depends("personas_ids")
    def _compute_total(self):
        for record in self:
            record.contador_personas_asignadas = len(record.personas_ids) if record.personas_ids else 0
